rules/Php80/NodeAnalyzer/PromotedPropertyCandidateResolver.php

```diff
@@ -46,12 +57,26 @@ public function resolveFromClass(Class_ $class, ClassMethod $constructClassMetho
                 continue;
             }
 
+            if (! $allowModelBasedClasses && $this->hasModelTypeCheck($property, 'JMS\Serializer\Annotation\Type')) {
```

Comment:
Better use class constant enum here , to keep classes separated and non-prefixed

Reply:
sure, updated to use class name constant enum  https://github.com/rectorphp/rector-src/pull/7043/commits/2ad2fd5755c31b401aa934e2f9cda4e6ead629d7 👍